src/lib/helperFunctions.ts

```diff
@@ -45,3 +45,25 @@ export const statusToNumber = (
   }
   return 0;
 };
+
+export const isHostnameLocal = (hostname: string) => {
+  if (hostname === "localhost" || hostname === "[::1]") {
+    return true;
+  }
+  // Test 127.0.0.0/8 range.
+  const blocks = hostname.split(".");
+  if (blocks.length === 4 && blocks[0] === "127") {
+    return blocks
+      .slice(1)
+      .every((block) => Number(block) >= 0 && Number(block) < 256);
```

Comment:
This can just `return true` — because 127.500.500.20` is not a valid IP, or alternatively:

```
if (LOCALHOST_IP_REGEX.test(hostname))
```

where `LOCALHOST_IP_REGEX` == `/^127.\d{1,3}.\d{1,3}.\d{1,3}$/`

Reply:
Mhh, the reason why I didn't go this way was because Firefox will parse `new URL("https://127.999.999.999")` (chromium doesn't) and hence `isUriLocalhost(uri: string)` would return true for that URI.

Reply:
You could still use a regex here instead of split/every

Reply:
You mean like this?
```typescript
  const LOCALHOST_IP_REGEX = /^127.(\d{1,3}).(\d{1,3}).(\d{1,3})$/;

  // Test 127.0.0.0/8 range.
  if (LOCALHOST_IP_REGEX.test(hostname)) {
    LOCALHOST_IP_REGEX.exec(hostname)
      ?.slice(1)
      .every((block) => Number(block) >= 0 && Number(block) < 256);
  }
  ```

Reply:
``` const LOCALHOST_IP_REGEX = /^127.(\d{1,3}).(\d{1,3}).(\d{1,3})$/;

  // Test 127.0.0.0/8 range.
  if (LOCALHOST_IP_REGEX.test(hostname)) {
    return true;
  }
```

Reply:
done with 5911a5375e455439c2e0c07bb1c6fd1742ab47d0